### PR TITLE
Ns 95/provide endpoint for network overview filters: added and tested endpoint for network overview filters

### PR DESCRIPTION
**Pull request by net-illia-stetsenko**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,38 +8,38 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
...
+
+* Rewrite ChartManager and DashboardManager. Get rid of multiple json converting
+
+* Rename requested_dashboard -> request_dashboard method
+
+* Change net-api version
+
+---------
+
 Co-authored-by: net-stalker-bot <githubbot@netstalker.io>
```
Comment (net-yehor-tretiakov) [inline CHANGELOG.md]: :D

File: net-gateway/src/main.rs [hunk 2]
```diff
@@ -15,12 +15,12 @@ async fn main() -> std::io::Result<()> {
             .wrap(
                 Cors::default()
                     // for profuction use this will be changed
-                    .allowed_origin("http://localhost:4000")
+                    .allowed_origin("http://localhost:3000")
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/main.rs]: Why exactly have you changed the port?

File: net-gateway/src/endpoints/charts/network_graph/request/manager.rs [hunk 3]
```diff
@@ -38,21 +37,22 @@ impl ChartRequestManagaer for NetworkGraphChartManager {
...
         #[allow(unused_variables)]
-        client_data: Arc<web::Query<ClientData>>
+        client_data: Arc<ClientData>,
+        filters: Option<Arc<Filters>>,
     ) -> Box<dyn API> {
         Box::new(NetworkGraphRequestDTO::new(
             params.start_date,
             params.end_date,
-            false
+            filters.as_ref().unwrap().as_ref().clone().into(),
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/charts/network_graph/request/manager.rs]: IDK, just a matter of codestyle (as for compiler this code will remain the same), but I prefer have such long line as its one variable instead.

File: net-gateway/src/endpoints/charts/network_bandwidth/request/manager.rs [hunk 4]
```diff
@@ -38,20 +37,22 @@ impl ChartRequestManagaer for NetworkBandwidthChartManager {
...
         #[allow(unused_variables)]
-        client_data: Arc<web::Query<ClientData>>
+        client_data: Arc<ClientData>,
+        filters: Option<Arc<Filters>>,
     ) -> Box<dyn API> {
         Box::new(NetworkBandwidthRequestDTO::new(
             params.start_date,
-            params.end_date
+            params.end_date,
+            filters.as_ref().unwrap().as_ref().clone().into(),
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/charts/network_bandwidth/request/manager.rs]: Same for here

File: net-gateway/src/endpoints/charts/network_bandwidth_per_endpoint/request/manager.rs [hunk 5]
```diff
@@ -38,20 +37,22 @@ impl ChartRequestManagaer for NetworkBandwidthPerEndpointChartManager {
...
         #[allow(unused_variables)]
-        client_data: Arc<web::Query<ClientData>>
+        client_data: Arc<ClientData>,
+        filters: Option<Arc<Filters>>,
     ) -> Box<dyn API> {
         Box::new(NetworkBandwidthPerEndpointRequestDTO::new(
             params.start_date,
-            params.end_date
+            params.end_date,
+            filters.as_ref().unwrap().as_ref().clone().into(),
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/charts/network_bandwidth_per_endpoint/request/manager.rs]: .

File: net-gateway/src/endpoints/charts/network_bandwidth/endpoint.rs [hunk 6]
```diff
@@ -10,28 +10,32 @@ use crate::authorization::Authorization;
...
     client_data: web::Query<ClientData>,
     params: web::Query<GeneralFilters>,
+    filters_wrapper: web::Query<FiltersWrapper>,
     req: HttpRequest,
 ) -> impl Responder {
     //Auth stuff
     if let Err(response) = Authorization::authorize(req, MockAuthenticator {}).await {
         return response;
     }
+    log::debug!("got request for network bandwidth chart, {:?}", filters_wrapper);
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/charts/network_bandwidth/endpoint.rs:32]: Plz, remove such logs

File: net-gateway/src/endpoints/filters/network_overview_filters/response/filters.rs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        serde_json::to_value(self).unwrap()
+    }
+
+    fn get_json_type(&self) -> &'static str {
+        JSON_TYPE
+    }
+}
+
+impl From<NetworkOverviewDashboardFiltersDTO> for NetworkOverviewFiltersResponse {
+    fn from(dto: NetworkOverviewDashboardFiltersDTO) -> Self {
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/filters/network_overview_filters/response/filters.rs]: I like not renaming trait method/fn fields. Plz rename `dto` -> `value` parameter.

**Review by net-yehor-tretiakov — APPROVED** [review]
After small takes, and `impl From<Filters> for` strange code, all lgtm. Plz make some changes before merge.

**And resolve all the conflicts with main

File: net-gateway/src/main.rs (hunk 2 above)
Comment (net-illia-stetsenko) [reply net-gateway/src/main.rs]: Our new react project usually runs on 3000 port

**Pull request merged**